### ISSUE TITLE
Implement ScheduleFree+ optimizer (large-batch-stable schedule-free)

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#5645]
## Summary

Add **ScheduleFree+** — the large-batch-stability variant of Schedule-Free — to
`src/odyssey/training/optimizers/` as a pure-functional step, mirroring the
existing first-order-moment optimizers (`adan.mojo`, `adopt.mojo`).

## Background

Base Schedule-Free (Defazio et al. 2024, "The Road Less Scheduled",
arXiv:2405.15682) removes explicit learning-rate schedules via online iterate
averaging over a fast sequence `z` and a running average `x`, evaluating the
gradient at an interpolated query point `y`. At LLM / large-batch scale the base
method can diverge, and its uniform averaging under-performs on long runs.

ScheduleFree+ (Defazio 2026, "ScheduleFree+: Scaling Learning-Rate-Free &
Schedule-Free Learning to Large Language Models", arXiv:2605.19095) adds three
stability mechanisms. The rule below is transcribed from the downstream tracking
issue (mvillmow/Random#78), whose wording is authoritative for this
implementation.

## Update rule

Per step `t` (1-indexed), with caller-supplied globally-reduced objective
`f(y_t)`:

1. **Inner momentum** (momentum inside the base-seq `z`-update):

   ```
   m_{t+1} = mu * m_t + (1 - mu) * g_t
   z_{t+1} = z_t - lr_t * m_{t+1}
   ```

2. **Polyak step-size**:

   ```
   gnorm_t = rho * gnorm_{t-1} + (1 - rho) * mean(|g_t|)          # L1-EMA
   polyak  = max(0, f(y_t) + beta_sf * <g_t, z_t - x_t>) / (gnorm_t + eps)
   lr_t    = learning_rate * polyak
   ```

3. **Increasing outer momentum** (`beta_out` annealed `beta_sf` -> `beta_max`):

   ```
   frac     = min(1, (t - 1) / max(1, horizon - 1))
   beta_out = beta_sf + (beta_max - beta_sf) * frac
   x_{t+1}  = beta_out * x_t + (1 - beta_out) * z_{t+1}
   y_{t+1}  = (1 - beta_out) * z_{t+1} + beta_out * x_{t+1}       # next query
   ```

Initial state: `z_1 = x_1 = params`, `m_1 = 0`, `gnorm_0 = 0`.

## Scope

- `schedule_free_plus_step` (full hyperparameters) and
  `schedule_free_plus_step_simple` (default hyperparameters), pure-functional
  (state in, state out) per the optimizer-module convention.
- Shape/dtype guards on `params`/`gradients` plus a `params`/momentum-buffer
  shape guard.
- float32 / float64 API (first-order); the two scalar reductions
  (`<g, z-x>`, `mean(|g|)`) read elementwise as Float64 and so work for any real
  params dtype.

## Acceptance criteria

- [ ] `schedule_free_plus.mojo` lands with both step functions and guards.
- [ ] Unit test: multi-step (3-step) numerical parity to 1e-9 against a NumPy
      reference generator, plus shape/dtype guards, `simple == full` defaults,
      and descent direction.
- [ ] Registered in `scripts/run_primitive_test.sh`, the README primitive table,
      and the `optimizers/__init__.mojo` re-export.
- [ ] `mojo format` clean; pre-commit clean on all touched files.

## References

- Defazio, A., Yang, X. A., Mehta, H., Mishchenko, K., Khaled, A., & Cutkosky,
  A. (2024). The Road Less Scheduled. NeurIPS 2024. arXiv:2405.15682.
- Defazio, A. (2026). ScheduleFree+: Scaling Learning-Rate-Free & Schedule-Free
  Learning to Large Language Models. arXiv:2605.19095.
